### PR TITLE
Adds default values for Counters and Lists fields

### PR DESCRIPTION
**Pull request by igooch**
**What type of PR is this?**
/kind feature

**What this PR does / Why we need it**:
Adds default values for Counters and Lists fields. 
Populates initial GamerServerStatus for Counters and Lists from the GameServerSpec.
Removes remaining references to featureGates in CRDs (these fields are validated in the validate function). 

**Which issue(s) this PR fixes**:
Working on #2716 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/apis/agones/v1/gameserver.go
Comment (gongmax) [inline pkg/apis/agones/v1/gameserver.go:455]: Is this logic needed? Does `gs.Status.Counters = gs.Spec.Counters` simply work? I'm not sure how the deepcopy works but I'm taking `gs.applyEvictionStatus()` as reference.

**Review by gongmax — APPROVED** [review]
LGTM

File: pkg/apis/agones/v1/gameserver.go
Comment (markmandel) [inline pkg/apis/agones/v1/gameserver.go:308]: Since it's used in both Spec and Status, maybe the names should be `Counter` and `List` ? WDYT?

**Review by markmandel — CHANGES_REQUESTED** [review]
Just marking as request changes, since I'd like to get my comments reviewed before merging.

**Review by markmandel — APPROVED** [review]
LGTM! (Assuming everything passes 👍🏻 )

**Comment by markmandel** [comment]
Ruh roh. Lots of conflicts. Sorry - that was my CRD work.

**Pull request merged**
